tests/test_salesforce_api_per_run_quota.py

```diff
@@ -0,0 +1,69 @@
+from sfbase import SFBaseTest
+from tap_tester import connections, runner, menagerie
+from tap_tester.logger import LOGGER
+
+
+class SFAPIQuota(SFBaseTest):
+    """
+    https://jira.talendforge.org/browse/TDL-23431
+    This testcase makes sure we are able to configure the tap with specific 
+    per_run quota for apis and verifies if tap is working as per the 
+    api quota limit set.
+    """
+
+    """
+    Set a start date well past so that we have enough data to sync to hit the limit
+    set a low quota to make the sync fail
+    some streams are excluded as they don't have any data 
+    """
+
+    start_date = '2000-11-23T00:00:00Z'
+    per_run_quota = '1'
+    streams_to_exclude = {
+             'DatacloudAddress',
+             'DatacloudCompany',
+             'DatacloudContact',
+             'DatacloudDandBCompany',
+             'DatacloudOwnedEntity',
+             'DatacloudPurchaseUsage',
+             'FieldSecurityClassification',
+             'ServiceAppointmentStatus',
+             'WorkOrderLineItemStatus',
+             'WorkOrderStatus',
+             'ShiftStatus',
+             'WorkStepStatus',
+        }
+
+    @staticmethod
+    def name():
+        return "tt_sf_api_quota"
+
+    def streams_to_test(self):
+        return self.expected_streams().difference( self.streams_to_exclude )
+
+    @staticmethod
+    def streams_to_selected_fields():
+        # Select all fields in the selected catalogs (streams)
+        return {}
+
+    def test_api_per_run_quota(self):
+        """
+        Run the tap in check mode and verify it returns the error for quota limit reached. If not, proceed to sync mode.
+        """
+        expected_per_run_error = "Terminating replication due to allotted quota"
+        conn_id = connections.ensure_connection(self)
+
+        found_catalogs = self.run_and_verify_check_mode( conn_id)
```

Comment:
Does this handle the case where check mode fails?

Reply:
I noticed that total quota check is caught in check mode and per run quota is caught in sync mode. I added that assertion as well to take care of the failure in check mode

Reply:
Are we sure that total is always only caught in check mode and per run in sync mode?  Can we prove this and document it in the test?

Reply:
Added the following comment in the test.
For the sync mode, we have a higher total quota set, so it is unlikely to hit the total quota. Noticed that per run quota limit reaches only during the sync mode.